roles/os_hardening/tasks/limits.yml

```
@@ -27,10 +27,45 @@
       modification_time: preserve
       access_time: preserve
 
-  when: not os_security_kernel_enable_core_dump | bool
+  when: ansible_service_mgr != "systemd" and not os_security_kernel_enable_core_dump | bool
 
 - name: Remove 10.hardcore.conf config file
   file:
     path: /etc/security/limits.d/10.hardcore.conf
     state: absent
-  when: os_security_kernel_enable_core_dump | bool
+  when: ansible_service_mgr != "systemd" and os_security_kernel_enable_core_dump | bool
```

If an application runs on a system with systemd but without using systemd, we want to enforce the coredump setting, too. So these tasks should always run. And then you can move this task down to the "remove"-block.

```suggestion
  when: not os_security_kernel_enable_core_dump | bool
```